lib/expand-ports.js

```diff
@@ -25,12 +23,24 @@ const expandPorts = ports => {
       throw new Error(typeof width);
     }
 
+    return {
+      direction: direction,
+      width: width
+    };
+  }
+};
+
+const expandPorts = ports => {
+  if (ports.constructor !== Object) {
+    return ports.foreach((port) => {
```

Comment:
Have you tested this case? I'm pretty sure that in JavaScript, `forEach` should have a capital `E`. I'm assuming that `ports` is an Array in this case.

Also, you can't just return a `ports.forEach()`, since the [return value of `.forEach()` is `undefined`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#Return_value). You either want to use a `.map()`, making sure that the inner function actually returns the mutated port, or you want to do the `ports.forEach()` without the `return` on one line and then return the mutated `ports` object on the next line.